updated secrets backend to localFileSystem.

atish-beehyv opened the pull request.

morrie-seesaw commented:
@greg-seesaw and @atish-beehyv  I made a few changes here:
- I added a mount on `docker/docker-compose-local.yml` where a user can place Airflow connections and their snowflake private key. This is set via in the `AIRFLOW_LOCAL_CONNECTIONS_DIR` and should be out of source control (to prevent pushing secrets/keys to github)
- Updated the airflow local file system backend to be reference `./files/variables.json` and `./files/connections.json`. I couldn't get Airflow to retrieve secrets with `${AIRFLOW_HOME}` as a prefix.
- Added an example connection for a Snowflake connection

When I put my connections.json and private key in a local directory and set `AIRFLOW_LOCAL_CONNECTIONS_DIR` to that local directory, I can then launch local runner and run the health check DAG.

atish-beehyv commented:
> @greg-seesaw and @atish-beehyv I made a few changes here:
> 
>     * I added a mount on `docker/docker-compose-local.yml` where a user can place Airflow connections and their snowflake private key. This is set via in the `AIRFLOW_LOCAL_CONNECTIONS_DIR` and should be out of source control (to prevent pushing secrets/keys to github)
> 
>     * Updated the airflow local file system backend to be reference `./files/variables.json` and `./files/connections.json`. I couldn't get Airflow to retrieve secrets with `${AIRFLOW_HOME}` as a prefix.
> 
>     * Added an example connection for a Snowflake connection
> 
> 
> When I put my connections.json and private key in a local directory and set `AIRFLOW_LOCAL_CONNECTIONS_DIR` to that local directory, I can then launch local runner and run the health check DAG.

Hey @morrie-seesaw Any new steps to add before local runner start?
For me steps which I was following is not working now and getting error
![image](https://github.com/shadowpuppet/aws-mwaa-local-runner/assets/156322430/69b36ab9-9ad2-49f4-afe3-f95190251208)

morrie-seesaw commented:
> > @greg-seesaw and @atish-beehyv I made a few changes here:
> > ```
> > * I added a mount on `docker/docker-compose-local.yml` where a user can place Airflow connections and their snowflake private key. This is set via in the `AIRFLOW_LOCAL_CONNECTIONS_DIR` and should be out of source control (to prevent pushing secrets/keys to github)
> > 
> > * Updated the airflow local file system backend to be reference `./files/variables.json` and `./files/connections.json`. I couldn't get Airflow to retrieve secrets with `${AIRFLOW_HOME}` as a prefix.
> > 
> > * Added an example connection for a Snowflake connection
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > When I put my connections.json and private key in a local directory and set `AIRFLOW_LOCAL_CONNECTIONS_DIR` to that local directory, I can then launch local runner and run the health check DAG.
> 
> Hey @morrie-seesaw Any new steps to add before local runner start? For me steps which I was following is not working now and getting error


@atish-beehyv try these steps:
- Create a local directory outside of the repo
- Add your Snowflake private key to this directory
- Using the example connections json file I added, create a snowflake connection with your credentials. Be sure to include `"./files/{name_of_private_key}.p8"`. Put this connection into the directory created in step 1.
- Open a shell session in the local runner repo
- Run `export AIRFLOW_LOCAL_CONNECTIONS_DIR=name_of_local_directory`. 
- Run `./mwaa-local-env start`
- Update the connection name in the health check dag to match the name is connections.json
- Run the health check DAG

I made a [small change to the docker compose file](https://github.com/shadowpuppet/aws-mwaa-local-runner/pull/2/files#diff-e4a9a85f862fd0fa0374401140c037eaf7ec2d7e48ddc1581a461d5889eebd11R37) to avoid the error you are seeing. If the environment variable `AIRFLOW_LOCAL_CONNECTIONS_DIR` is not set, the local runner will default to  the `example_connections` directory in the repo.

atish-beehyv commented:
> > > @greg-seesaw and @atish-beehyv I made a few changes here:
> > > ```
> > > * I added a mount on `docker/docker-compose-local.yml` where a user can place Airflow connections and their snowflake private key. This is set via in the `AIRFLOW_LOCAL_CONNECTIONS_DIR` and should be out of source control (to prevent pushing secrets/keys to github)
> > > 
> > > * Updated the airflow local file system backend to be reference `./files/variables.json` and `./files/connections.json`. I couldn't get Airflow to retrieve secrets with `${AIRFLOW_HOME}` as a prefix.
> > > 
> > > * Added an example connection for a Snowflake connection
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > When I put my connections.json and private key in a local directory and set `AIRFLOW_LOCAL_CONNECTIONS_DIR` to that local directory, I can then launch local runner and run the health check DAG.
> > 
> > 
> > Hey @morrie-seesaw Any new steps to add before local runner start? For me steps which I was following is not working now and getting error
> 
> @atish-beehyv try these steps:
> 
>     * Create a local directory outside of the repo
> 
>     * Add your Snowflake private key to this directory
> 
>     * Using the example connections json file I added, create a snowflake connection with your credentials. Be sure to include `"./files/{name_of_private_key}.p8"`. Put this connection into the directory created in step 1.
> 
>     * Open a shell session in the local runner repo
> 
>     * Run `export AIRFLOW_LOCAL_CONNECTIONS_DIR=name_of_local_directory`.
> 
>     * Run `./mwaa-local-env start`
> 
>     * Update the connection name in the health check dag to match the name is connections.json
> 
>     * Run the health check DAG
> 
> 
> I made a [small change to the docker compose file](https://github.com/shadowpuppet/aws-mwaa-local-runner/pull/2/files#diff-e4a9a85f862fd0fa0374401140c037eaf7ec2d7e48ddc1581a461d5889eebd11R37) to avoid the error you are seeing. If the environment variable `AIRFLOW_LOCAL_CONNECTIONS_DIR` is not set, the local runner will default to the `example_connections` directory in the repo.

thanks it worked for me too.
So I removed connections and variables json file from docker directory which I added in previous commits. We will read from local system instead of airflow home directory.

greg-seesaw commented on example_connections/example_snowflakle_connections.sql:
I think this should have extension `.json`

greg-seesaw reviewed (approved):
I can now connect from my local Docker to Snowflake too. Thank you!

I think this is ready to merge once the example connection file is updated.

I've also been updating the documentation here: https://docs.google.com/document/d/1MnZFE__w_tHc2ONpfy21Fm3D6zfjzL6yRe_RDMg5bwo/edit?usp=sharing

Please update as you see fit.
